src/components/page/FoodsPage.tsx

```diff
@@ -1,3 +1,156 @@
+  const [categoryList, setCategoryList] = useState<string[]>(
+    searchParams?.get('category')?.split(',') || [],
+  );
+
+  const containersGroupByGroups = groupContainersByGroupId(containers);
+  const containerIdMap = createContainerIdNameMap(containers);
+  const groupIdMap = createGroupIdNameMap(containers);
+  const updateUrlParams = (key: string, value?: string | string[]) => {
+    if (!value || (Array.isArray(value) && value.length === 0)) {
+      params.delete(key);
+    } else {
+      params.set(key, Array.isArray(value) ? value.join(',') : value);
+    }
+    replace(`${pathname}?${params.toString()}` as Route);
+  };
+
+  const removeGroupFilter = () => {
+    updateUrlParams('group');
+  };
+
+  const removeContainerFilter = () => {
+    updateUrlParams('container');
+  };
+
+  const removeCategoryFilter = (key: string) => {
+    const updatedCategoryList = categoryList.filter((category) => category !== key);
+    setCategoryList(updatedCategoryList);
+    updateUrlParams('category', updatedCategoryList);
+  };
+
+  useEffect(() => {
+    const categoryList = searchParams?.get('category')?.split(',') || [];
+    setCategoryList(categoryList);
+    setQuery(searchParams?.get('query') || '');
+
+    const filterByGroup = (row: { groupId: string }) => group === '' || group === row.groupId;
+    const filterByContainer = (row: { id: string }) => container === '' || container === row.id;
+    const filterByCategory = (food: { category: string }) => {
+      if (!categoryList.length) return true;
+      return categoryList.some((c) => food.category.includes(c));
+    };
+    const filterByName = (food: { name: string }) =>
+      food.name.toLowerCase().includes(query.toLowerCase());
+
+    const filteredContainers = containers.filter(filterByGroup).filter(filterByContainer);
+    const initialFoodsView: IFoodView[] = filteredContainers.flatMap((container: IContainer) =>
+      container.foods.map((food: IFood) => ({
+        ...food,
+        container: container.name,
+      })),
+    );
+    const filteredFoods = initialFoodsView.filter(filterByName).filter(filterByCategory);
+
+    const sortFoods = (a: IFood, b: IFood) => {
+      switch (sort) {
+        case 'name':
+          return a.name.localeCompare(b.name);
+        case 'expiry':
+          return a.expiry < b.expiry ? -1 : 1;
+        default:
+          return a.createdAt > b.createdAt ? -1 : 1; // default is createdAt desc order
+      }
+    };
+
+    const sortedFoods = [...filteredFoods].sort(sortFoods);
+    setDisplayedFoods(sortedFoods);
+  }, [containers, query, sort, group, container, searchParams]);
+
+  return (
+    <div className="mt-6 mx-4 mb-16">
+      <div className="relative">
+        <SearchBar />
+        <FilterButton
+          containers={containersGroupByGroups}
+          containerIdMap={containerIdMap}
+          groupIdMap={groupIdMap}
+        />
+      </div>
+      <div
+        className={`flex ${
```

Comment:
### (Optional)
The code here is having no problem, my suggestion here is just optional !

I often like using twMerge(in our project, we can utilize `cn()` method from `@/lib/tailwind/utils`) to avoid using template literals inside of className. 
Because, when you're using template literal inside of className, just missing a single space can easily cause an unexpected behaviour.

If you use `cn()` method in here, it would be like this ↓
`   className={cn(
          'flex',
          group ||
            container ||
            (categoryList.length > 0 &&
              'gap-1.5 mt-4 mb-1.5 sm:overflow-auto sm:whitespace-normal overflow-scroll whitespace-nowrap'),
        )}`
      
This is not the main purpose of using `twMerge` tho, plz take a look and consider it 🙏

Reply:
After experimenting, I figured out that giving the following styles works perfectly
```
    margin-left: -16px;
    margin-right: -16px;
    padding-left: 16px;
    padding-right: 16px;
    padding-bottom: 4px;
```

Reply:
> The badges' text breaks unwantedly when the screen width increases. Could you investigate whether the cause is ascribed to this PR. We might need a separate issue to handle this though.

I created issue https://github.com/genesis-tech-tribe/nishiki-frontend/issues/117, I will investigate this with this ticket

Reply:
Finally, I set these css for this. Could you please check at your enviroment? 
I thought #117 is also solved. 

> flex -mx-4 px-4 gap-1.5 mt-4 pb-1 overflow-x-auto whitespace-nowrap

Reply:
I appreciate it!